File: src/SaruchSDK/Interfaces/IAccount.vb
Imports SaruchSDK.JSON

Public Interface IAccount

    ''' <summary>
    ''' user metadata
    ''' </summary>
    Function UserInfo() As Task(Of JSON_UserInfo)
    ''' <summary>
    ''' account storage info
    ''' </summary>
    Function StorageInfo() As Task(Of JSON_StorageInfo)
    ''' <summary>
    ''' renew authentication token
    ''' </summary>
    Function RenewToken() As Task(Of JSON_GetToken)
    ''' <summary>
    ''' get video Conversion Settings
    ''' </summary>
    Function GetConversionSettings() As Task(Of JSON_GetConversionSetings)
    ''' <summary>
    ''' change video Conversion Settings
    ''' </summary>
    Function ChangeConversionSettings(Resolutions As Dictionary(Of utilitiez.ResolutionEnum, Boolean)) As Task(Of Boolean)
End Interface

File: SaruchSDK/SaruchSDK/Interfaces/IAccount.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using SaruchSDK.JSON;

namespace SaruchSDK
{
    public interface IAccount
    {
        Task<bool> ChangeConversionSettings(Dictionary<Utilitiez.ResolutionEnum, bool> Resolutions);
        Task<JSON_GetConversionSetings> GetConversionSettings();
        Task<JSON_GetToken> RenewToken();
        Task<JSON_StorageInfo> StorageInfo();
        Task<JSON_UserInfo> UserInfo();
    }
}